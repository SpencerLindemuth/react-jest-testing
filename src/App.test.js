import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import NavBar from './components/navbar'
import * as TestRenderer from 'react-test-renderer'


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders a component called NavBar', () => {
  let testRenderer = TestRenderer.create(<NavBar />);
  expect(testRenderer.toJSON()).toMatchSnapshot();
})

it('has a NavBar component with text "Site Logo Here"', () => {
  let renderer = TestRenderer.create(<NavBar />);
  let instance = renderer.root
  let node = instance.find(node => node.type === Child).props.children.type === 'p';
  expect(node.innerText).toBe("Site logo here")
})
