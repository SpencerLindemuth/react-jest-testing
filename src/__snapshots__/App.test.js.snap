// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a component called NavBar 1`] = `
<div
  id="navbar"
>
  Site logo here
</div>
`;
